File: Learning/OtherLanguages/C/C练习题/Use Python/16.py
#辗转相除法

def gcd(a, b):
    if a % b != 0:
        return gcd(b, a%b)
    else:
        return b
    
print('请输入两个数字：')
m = int(input('第一个数:'))
n = int(input('第二个数:'))
if m < n:
    temp = m
    m = n
    n = temp
print('最大公约数是：', gcd(m, n))
print('最小公倍数是：', m*n //gcd(m,n))

File: Learning/OtherLanguages/C/C练习题/Use C/16.c
#include <stdio.h>


void main()
{
    int a, b, num1, num2, temp;

    printf("Input two number:");
    scanf("%d%d", &num1, &num2);

    a = num1>num2 ? num1 : num2;
    b = num1<num2 ? num1 : num2;

    while(b!=0)   //辗转相除法
    {
        temp = a % b;
        a = b;
        b = temp;
    }
    printf("最大公约数：%d\n", a);
    printf("最小公倍数：%d\n", num1*num2/a);

    system("pause");
}
